SWLOR.Game.Server/Service/Communication.cs

```diff
@@ -401,34 +419,56 @@ public static void ProcessChatMessage()
                 var g = (byte)(color >> 16 & 0xFF);
                 var b = (byte)(color >> 8 & 0xFF);
 
+
+                var langMessageForCombatLog = new StringBuilder();
+
                 if (language != SkillType.Basic)
                 {
                     var languageName = Language.GetName(language);
-                    finalMessage.Append(ColorToken.Custom($"[{languageName}] ", r, g, b));
+                    string cToken = ColorToken.Custom($"[{languageName}] ", r, g, b);
+                    finalMessage.Append(cToken);
+                    langMessageForCombatLog.Append(ColorToken.Custom(GetName(sender), 153, 255, 255));
+                    langMessageForCombatLog.Append(": ");
+                    langMessageForCombatLog.Append(cToken);
                 }
 
+                bool minRank = false;
+
                 foreach (var component in chatComponents)
                 {
+                    string languageText = "";
                     var text = component.Text;
+                    var textForCombatLog = component.Text;
 
                     if (component.IsTranslatable && language != SkillType.Basic)
                     {
-                        text = Language.TranslateSnippetForListener(sender, obj, language, component.Text);
+                        bool maxRank = false;
```

Comment:
No need to declare this here. Just do it on the next line (446)

out maxRank -> out var maxRank